aquarius/events/events_monitor.py

```diff
@@ -574,70 +505,105 @@ def get_assets_in_chain(self):
-            hash_text = "DispenserCreated(address,address,uint256,uint256,address)"
-        elif event_name == EventTypes.EVENT_TRANSFER:
-            hash_text = "Transfer(address,address,uint256)"
-        else:
-            hash_text = (
-                "OrderStarted(address,address,uint256,uint256,uint256,address,uint256)"
-            )
-
-        event_signature_hash = self._web3.keccak(text=hash_text).hex()
-
-        _from = from_block
-        _to = min(_from + chunk_size - 1, to_block)
-
-        logger.debug(
-            f"Searching for {event_name} events on chain {self._chain_id} "
+    def get_event_logs(self, from_block, to_block):
+        """Get all events from -> to and process them.
+        If that fails, and we tried with multiple blocks, let split handle it
+        If that fails, and we tried on a single block, then try to get events one by one instead of all
+        """
+        logger.info(
+            f"Searching for events events on chain {self._chain_id} "
             f"in blocks {from_block} to {to_block}."
         )
 
         filter_params = {
-            "topics": [event_signature_hash],
-            "fromBlock": _from,
-            "toBlock": _to,
+            "topics": [list(EventTypes.hashes.keys())],
+            "fromBlock": from_block,
+            "toBlock": to_block,
         }
 
-        all_logs = []
-        while _from <= to_block:
-            # Search current chunk
+        try:
             logs = self._web3.eth.get_logs(filter_params)
-            all_logs.extend(logs)
-            if (_from - from_block) % 1000 == 0:
-                logger.debug(
-                    f"Searched blocks {_from} to {_to} on chain {self._chain_id}; "
-                    f"{len(all_logs)} {event_name} events detected so far."
-                )
+        except Exception as e:
+            if from_block < to_block:
+                # splitting in two might help, so rely on that
+                raise Exception(f"Failed to get events for multiple blocks")
+            else:
+                # Since there is only one block, and we failed to get all events, we need to try to take them one by one
+                # if any call fails, there is nothing more we can do  (ie:  failed to get only transfer events from block X)
+                for topic in EventTypes.hashes:
+                    filter_params = {
+                        "topics": [topic],
+                        "fromBlock": from_block,
+                        "toBlock": to_block,
+                    }
+                    try:
+                        logs = self._web3.eth.get_logs(filter_params)
+                        self.process_logs(logs, to_block)
+                    except Exception as e:
+                        logger.error(
+                            f"Failed to fetch {EventTypes.hashes[topic]['type']} logs from block {from_block}.  Bailing out.."
+                        )
+                return
+        try:
```

Comment:
if this function also does the processing, I would suggest `get_and_process_logs`, otherwise for me it would look like this function call belongs in process_block_range after the retrieval. Either way is fine, but let's name it accordingly.

Reply:
no, it's only processing logs.
I have added more function descriptions

Reply:
so can we rename the function get_and_process_logs, since get_event_logs also does processing?